frontend/src/metabase/core/components/TabList/TabList.tsx

```diff
@@ -23,20 +27,72 @@ const TabList = forwardRef(function TabGroup<T>(
 ) {
   const idPrefix = useUniqueId();
   const outerContext = useContext(TabContext);
+
+  const [scrollPosition, setScrollPosition] = useState(0);
+  const [showScrollLeft, setShowScrollLeft] = useState(false);
+  const [showScrollRight, setShowScrollRight] = useState(false);
+
+  const tabListContentRef = useRef(null);
+
   const innerContext = useMemo(() => {
     return { value, idPrefix, onChange };
   }, [value, idPrefix, onChange]);
+
   const activeContext = outerContext.isDefault ? innerContext : outerContext;
 
+  const scroll = (direction: string) => {
+    if (tabListContentRef.current) {
+      const container = tabListContentRef.current as HTMLDivElement;
+
+      const scrollDistance =
+        (container.offsetWidth - 32) * (direction === "left" ? -1 : 1);
+      container.scroll(container.scrollLeft + scrollDistance, 0);
+      setScrollPosition(container.scrollLeft + scrollDistance);
+    }
+  };
+
+  useEffect(() => {
+    if (!tabListContentRef.current) {
+      return;
+    }
+
+    const container = tabListContentRef.current as HTMLDivElement;
+    setShowScrollRight(
+      scrollPosition + container.offsetWidth < container.scrollWidth,
+    );
+    setShowScrollLeft(scrollPosition > 0);
+  }, [setShowScrollLeft, setShowScrollRight, scrollPosition]);
```

Comment:
setters from `useState` will never change, so it's safe to omit them from hook dependencies.
```suggestion
  }, [scrollPosition]);
```